### PR TITLE
Migrate to "Jenkins Pipeline Unit testing framework"

### PR DESCRIPTION
**Pull request by ioboi**
This pull request is a demonstration if we decide to move from Spock to JenkinsPipelineUnit.

**Note: Only migrated tests work at the moment.**

This pull request is intended as a discussion basis.

### CODE REVIEW

File: test/groovy/vars/CheckSpec.groovy [hunk 1]
```diff
@@ -1,28 +1,46 @@
 package groovy.vars
 
-import com.homeaway.devtools.jenkins.testing.JenkinsPipelineSpecification
+import com.lesfurets.jenkins.unit.BasePipelineTest
+import org.junit.Before
+import org.junit.Test
+import static org.assertj.core.api.Assertions.*;
```
Comment (SylivanKenobi) [inline test/groovy/vars/CheckSpec.groovy:6]: What is the reason for using this import instead of the junit assertions?

**Comment by chrira** [comment]
Tests seem more object oriented -> good
I miss the test method structure "given, when, then". Could be done with comments.

What are the benefits for changing the test framework?
Are all tests easy to migrate?

**Comment by ioboi** [comment]
Hi @chrira 

The only benefit I see at the moment is, that we migrate to another Unit-Test-Framework (JUnit) but this is also just a matter of taste I guess :smile: 

As most tests at the moment only happen to check the call stack, we would have to write a lot of code.

For example before:

````
1 * getPipelineMock("error").call('missing parameter: hallo')
````

this becomes:

````
assertThat(helper.callStack.stream()
                 .filter({ it.methodName == "error" })
                .flatMap({ Arrays.stream(it.args) })
                .any({ it == "missing parameter: hallo" })).isTrue()
````

with AssertJ or with *just* JUnit:

````
assertTrue helper.callStack.stream()
                 .filter({ it.methodName == "error" })
                .flatMap({ Arrays.stream(it.args) })
                .any({ it == "missing parameter: hallo" })
````

As I understand Jenkins Pipeline Unit really shines when used to test pipelines using the shared library.

**Review by chrira — APPROVED** [review]
thanks

**Pull request merged**